### PR TITLE
Added new method to check token expiration

### PR DESCRIPTION
**Pull request by nipampatel**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -296,7 +296,15 @@ class Provider extends EventEmitter
                             expiry : token_details.expiry });
                     }.bind(this)).catch(reject);
                 });
-            }
+			},
+			is_token_within_seconds: (seconds) => {
+                return new Promise((resolve, reject) => {
+                        this.auth.get_token_details(token).then(function(token_details) {						
+							let time_diff_seconds = (Date.now() - token_details.iat) / 1000;
+							resolve(time_diff_seconds >  seconds);
```
Comment (svtokarevsv) [inline lib/index.js]: Your method name mismatches what it returns:
is_token_within_seconds - means that token is less or equal to seconds passed. While your resolve(time_diff_seconds >  seconds); 
means the opposite.

File: test/test.js [hunk 2]
```diff
@@ -457,6 +457,92 @@ describe("Provider", function()
...
+            data.should.eql({id: "Hello world"});
+
+            this.should.have.property("context");
+
+            var context = this.context();
+            should.exist(context);
+            context.should.have.property("token_details");
+
+            let waitTill = new Date(new Date().getTime() + 11 * 1000);
+            while(waitTill > new Date()){}
```
Comment (svtokarevsv) [inline test/test.js]: See my comment in your PR to order.cdl

File: lib/index.js (hunk 1 above)
Comment (jpdemiranda) [reply lib/index.js]: agree

File: lib/index.js [hunk 3]
```diff
@@ -296,7 +296,15 @@ class Provider extends EventEmitter
                             expiry : token_details.expiry });
                     }.bind(this)).catch(reject);
                 });
-            }
+			},
+			is_token_within_seconds: (seconds) => {
+                return new Promise((resolve, reject) => {
```
Comment (jpdemiranda) [inline lib/index.js]: Can you fix the formatting here

File: test/test.js [hunk 4]
```diff
@@ -457,6 +457,92 @@ describe("Provider", function()
...
+        
+        let new_id = ID("user","cdl","user","simulateduser1234");
+        provider.on("/helloWorld", function(data, next)
+        {
+            should.exist(data);
+            data.should.eql({id: "Hello world"});
+
+            this.should.have.property("context");
+
+            var context = this.context();
```
Comment (jpdemiranda) [inline test/test.js]: let please

File: test/test.js [hunk 5]
```diff
@@ -457,6 +457,92 @@ describe("Provider", function()
...
+            should.exist(context);
+            context.should.have.property("token_details");
+
+            let waitTill = new Date(new Date().getTime() + 11 * 1000);
```
Comment (jpdemiranda) [inline test/test.js]: why are we still waiting to time to pass?

File: test/test.js [hunk 6]
```diff
@@ -457,6 +457,92 @@ describe("Provider", function()
...
+                should.exist(res);
+                res.should.have.property("result").eql("dummy data");
+                done();
+            });
+        });
+    });
+    
+    it("should check token is not expired within given seconds", function(done)
+    {
+        this.test.timeout(20000);
```
Comment (jpdemiranda) [inline test/test.js]: Test should not take this long

File: lib/index.js (hunk 1 above)
Comment (nipampatel) [reply lib/index.js]: Thanks @svtokarevsv ! Nice catch. Fixed.

File: test/test.js (hunk 2 above)
Comment (nipampatel) [reply test/test.js]: @svtokarevsv Thanks ! Fixed with async await and settimeout().

File: lib/index.js (hunk 3 above)
Comment (nipampatel) [reply lib/index.js]: @jpdemiranda Thanks ! Formatting changed. I am not sure what keeps changing it in editor.

File: test/test.js (hunk 4 above)
Comment (nipampatel) [reply test/test.js]: @jpdemiranda replaced with let.

File: test/test.js (hunk 6 above)
Comment (nipampatel) [reply test/test.js]: @jpdemiranda Reduce run time with lower expiration limits. Please review.

File: test/test.js (hunk 5 above)
Comment (nipampatel) [reply test/test.js]: @jpdemiranda For "should check token is within given seconds" we don't need to wait but for "should check token is not within seconds" we have to wait till token expires. Please let me know if I missed something.

**Comment by nipampatel** [comment]
@jpdemiranda @svtokarevsv Thanks for review ! Requested changes done. Can you please review again?

**Pull request merged**